src/libraries/System.Linq/src/System/Linq/Grouping.cs

```diff
@@ -411,11 +409,7 @@ public IEnumerator<TElement> GetEnumerator()
         void ICollection<TElement>.CopyTo(TElement[] array, int arrayIndex) =>
             Array.Copy(_elements, 0, array, arrayIndex, _count);
 
-        bool ICollection<TElement>.Remove(TElement item)
-        {
-            ThrowHelper.ThrowNotSupportedException();
-            return false;
-        }
+        bool ICollection<TElement>.Remove(TElement item) => ThrowHelper.ThrowNotSupportedException_Boolean();
```

Comment:
That's the first time I notice throw helpers with bespoke return types. What are the benefits of doing it that way apart from brevity and could it create problems in a hypothetical toolchain that admits tail call optimization?

Reply:
I'm not aware of any problems this would lead to. It's also pre-existing this change; this is just using the existing helper in one more place like others it's already being used.